pawn-items.lic

```diff
@@ -0,0 +1,84 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#pawn-items
+=end
+
+custom_require.call %w[common common-items common-travel equipmanager]
+
+class PawnItems
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'source', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'Source container' },
+        { name: 'noun', regex: /^[A-z\.\s\-]+$/i, optional: true, variable: true, description: 'If specified, only items with this noun will be sold.' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+
+    # Put away items in hands to mitigate accidentally selling them
+    EquipmentManager.new.return_held_gear
+    # Warn the script user that this is designed to sell items off your person
+    warn_before_sell(args.source, args.noun)
+    # Turn items into coins
+    sell_items(args.source, args.noun)
+  end
+
+  def sell_items(source, noun)
+    # If container is very full then LOOK may not list all of them.
+    # If you're moving a specific item, then sort those to the top
+    # to increase chances we find and sell all of them in one go.
+    DRC.bput("sort #{noun} in my #{source}", "are now at the top", "What were you referring to", "Please rephrase that command", "You may only sort items in your inventory") if noun
+    DRCI.get_item_list(source, 'look')
+      .map { |full_name| full_name =~ /lot of other stuff/ ? full_name : full_name.split(' ').last }
+      .select { |item| noun ? /\b#{noun}\b/ =~ item : true }
+      .each do |item|
+        # This indicates there is more items than LOOK can show
+        # and we've reached the end of what we last saw.
+        # We need to look again to see what we can now see.
+        # Keep doing this until exhaust source.
+        if item =~ /lot of other stuff/
+          sell_items(source, noun)
+          return
+        end
+        # Attempt to get the item from the source container.
+        if DRCI.get_item?(item, source)
+          unless DRCT.sell_item(Room.current.id, item)
```

Comment:
Why was this helper introduced to `DRCT` when you only call it with the current room ID?

Reply:
I followed the example of [DRCT.buy_item(room_id, item)](https://github.com/rpherbig/dr-scripts/blob/c8e20343d9545637daaebec245646b3e7269f069/common-travel.lic#L35 method when I added `sell_item`. Both methods will walk you to the merchant to do the buy/sell. 

But for why did I do `Room.current.id` .... maybe I thought there'd be more than pawn shops you could sell items to? This probably should pull from the base town data to go to the _pawn shop_ considering the name and intent of this script, and support a town override argument (if you're in Crossing and want to pawn in Shard or Fang Cove)

Reply:
Resolved in https://github.com/rpherbig/dr-scripts/pull/5434/commits/ed46457064b60f222f6de0079fa9ebceacbf7f2c